### ISSUE TITLE
RAM usage

### ISSUE THREAD

**Issue by tudor-pop** [rsms/fb-mac-messenger#389]
I'm not sure if it's only me, but I'm kinda concerned about the amount of ram it eats. I mean Intellij IDEA or Android Studio eats about 1-1.5k RAM while Chrome with lots of tabs opened also eats about 1k. This app, doing nothing fancy (like debugging, compiling, autocomplete and what not) eats constantly around 900MB. 
macOS Sierra 10.12.4

**Comment by m4grio** [comment]
Came here to open the same issue.
It was consuming 900MB also, but after upgrading through `brew cask reinstall messenger` dropped to around 200MB – which is acceptable but still a bit high compared with other messaging apps that consume less, like WhatsApp around 140MB.
macOS Sierra 10.12.4 (16E195)

**Comment by tudor-pop** [comment]
Glad to hear that! I will try it later today and confirm any changes

**Comment by rsms** [comment]
When you say 1k @PopTudor, are you taking about kilobytes? If so, you are gravely mistaken — Chrome uses at least a couple of hundred megabytes for it's supervisor "main" process and a lot more memory if you combine the memory usage of it's many "render" processes (Chrome and Safari uses a pool of separate processes, each with its own resource usage.)

This app uses a single-process webview, meaning that this app does _not_ use additional processes to support the webview that loads messenger.com and in effect the sum of resource usage of this app (Messenger for Mac) needs to be compared to the _sum_ of memory usage by Chrome or Safari.

For instance, here's the memory usage of Chrome on my machine:
<img width="557" alt="screenshot 2017-04-26 10 38 36" src="https://cloud.githubusercontent.com/assets/47046/25448309/e242219e-2a6c-11e7-93ff-80d1bb554993.png">
If you sum the "Memory" column up, you get 2.6GB (2628MB)

Looking at this app's total memory usage, it's 360MB which includes the app itself + messenger.com in a webview:
<img width="514" alt="screenshot 2017-04-26 10 38 52" src="https://cloud.githubusercontent.com/assets/47046/25448406/3163f464-2a6d-11e7-8cfe-9aee0fb8e1a4.png">

So, not an issue, not a bug and incorrect assumptions about what is "normal" memory usage :)

**Comment by tudor-pop** [comment]
@m4grio I tried it but with no effect :(

@rsms I was talking in MB since I have not seen any app uning 1 kilobyte of RAM. It might be an incorrect assumption but from my point of view it should not eat so much. For the comparison I installed another Messanger app called Goofy (no affiliation with them) and here is the result: [
![image](https://cloud.githubusercontent.com/assets/4589309/26027007/207d97a4-380f-11e7-8b28-8944867a42df.png)
](url)
A couple of days ago it got to 1,8GB then I quit it. Don't get me wrong, I appreciate the effort and the software. Thank you!
As a guidance, I observed the memory starts to accumulate on scrolling up into the history and maybe it's not releasing previous resources. I don't know if you have control over that but it's a starting point :)
